kodi_addon_checker/check_files.py

```diff
@@ -131,3 +132,33 @@ def check_file_permission(report: Report, file_index: list):
         file = os.path.join(file["path"], file["name"])
         if os.path.isfile(file) and os.access(str(file), os.X_OK):
             report.add(Record(PROBLEM, "%s is marked as stand-alone executable" % relative_path(str(file))))
+
+
+def check_license_identifier(report: Report, addon_path: str):
+    """Check whether all the files present in the addon
+       contains the SPDX license header or not
+        :addon_path: Path of the addon
+    """
+
+    files = Path(addon_path).glob('**/*.py')
+
+    for file in files:
+        with open(str(file), 'r') as f:
+            tree = ast.parse(f.read())
+
+        docstring = ast.get_docstring(tree)
+        if docstring and "SPDX-License-Identifier" not in docstring:
+            report.add(Record(WARNING, "SPDX-License-Identifier is missing from addon file %s" %
+                              relative_path(str(file))))
+
+
+def check_license_tag(report: Report, parsed_xml):
+    """Check license tag in addon xml file
+        :parsed_xml: parsed tree for xml file
+    """
+
+    license_text = parsed_xml.find("extension/license").text
+    spdx_string = "GNU General Public License v2.0 or later"
```

Comment:
This forces every add-on to be `GPLv2+`.
I don't think that was your intention.

Reply:
@Rechi There's a problem with the identification of Licenses, lot's of addon have `GNU GENERAL PUBLIC LICENSE Version 3, 29 June 2007`  this type of text in their `license` tag. Now should I follow the identifiers in https://spdx.org/licenses/, doing that will generate an error on the type I mentioned above or should I just check that the license tag is non-empty, but doing this will pass all types of Licenses including any random string in the license tag.

One solution could be that we create a dictionary of all the license that we need and then use it.

Reply:
As I already mentioned at another PR, first require the licence tag to be listed exactly once and have a non empty string as value.
Also there is no reason to disallow non standard licences.